src/Time/Bool.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE AllowAmbiguousTypes        #-}
+{-# LANGUAGE ConstraintKinds            #-}
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE ExplicitForAll             #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE InstanceSigs               #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeApplications           #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeInType                 #-}
+{-# LANGUAGE TypeOperators              #-}
+{-# LANGUAGE UndecidableInstances       #-}
+
+module Time.Bool
+       ( And
+       , If
+       , type (==%)
+       ) where
+
+import Data.Type.Equality (type (==))
+
+import Time.Rational (type (:%), Rat)
+
+-- | Type-level 'if'.
+type family If (x :: Bool) a b where
+    If 'True  a _ = a
+    If 'False _ b = b
+
+-- | Type-level 'and'.
+type family And (a :: Bool) (b :: Bool) where
```

Comment:
Could you also write kinds of type family result, not only arguments? This kind can be inferred. But I think that it's better to write it explicitly, like we write types of arguments and type of result for usual functions.